krabs/krabs/filtering/event_filter.hpp

```diff
@@ -123,9 +142,14 @@ namespace krabs {
             return;
         }
 
-        if (!predicate_(record)) {
-            return;
-        }
+		if(predicate_ != nullptr) //only if predicate exists... else trust the native filtering
```

Comment:
Collapse the secondary check into an `&&` clause.

```suggestion
                //only if predicate exists... else trust the native filtering
		if (predicate_ && !predicate_(record)) {
		    return;
		}
```